lab/lab-1.md

```diff
@@ -0,0 +1,209 @@
+# Lab
+
+## Build backend
+
+Navigate to your user home directory or whereever you want to store the source code of the backend serivce.
+Now checkout the sample repository on your `node-0` via:
+
+```sh
+[centos@node-0 ~]$ cd
+[centos@node-0 ~]$ git clone --branch v1.0.0 https://github.com/hulk0301/example-app-backend.git
+[centos@node-0 ~]$ cd example-app-backend
+[centos@node-0 example-app-backend]$
+```
+
+Then build the docker image using the provided Dockerfile.
+Use `backend:v1.0.0` as tag for your image and push it to DockerHub(you should have already an account, if not created one).
+
+> Note: If you clone the repo to another folder, you might have to adapt some paths when running the provided commands!
+
+## Deploy the backend to Kubernetes
+
+### Create a new namespace
+
+We want to run our app in an isolated environment, therefore create a new namespace called `demoapp`:
+
+```sh
+[centos@node-0 ~]$ kubectl create namespace demoapp
+```
+
+### Deployment manifest
+
+Create a deployment manifest(`deployment_v1.yaml`) and apply it on the Kubernetes cluster using `kubectl`. Don't forget to specify some meaningful labels:
+
+- `app=backend`
+- `version=v1`
+
+Put the deployment into the newly created namespace `demoapp`:
+
+```sh
+[centos@node-0 example-app-backend]$ kubectl apply -f deployment_v1.yaml
```

Comment:
```suggestion
[centos@node-0 example-app-backend]$ kubectl apply -f deployment_v1.yaml -n demoapp
```

Reply:
Or add an instruction that this should be added to the manifest.

Reply:
done